src/ApparentHorizons/YlmSpherepack.hpp

```diff
@@ -0,0 +1,231 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "YlmSpherepackStorage.hpp"
+
+/// \ingroup Spectral
+/// \brief
+/// C++ interface to SPHEREPACK.
+class YlmSpherepack {
+ public:
+  using first_deriv_t = tnsr::i<DataVector, 2, Frame::Logical>;
+  using second_deriv_t = tnsr::ij<DataVector, 2, Frame::Logical>;
+  using interpolation_info_t = std::vector<std::vector<double>>;
+
+  /// Here l_max and m_max are the largest fully-represented l and m in
+  /// the Ylm expansion.
+  YlmSpherepack(size_t l_max, size_t m_max) noexcept;
+
+  bool operator==(const YlmSpherepack& rhs) const noexcept;
```

Comment:
`operator==` is symmetric so should be implemented as a free function instead of a member function.

Reply:
Even if I need to make more public member functions (or use `friend`) to implement a free operator==?

Reply:
(In this case I don't need to use `friend` but what if I did?)